### PR TITLE
Feature/an 2016 timeline cards overflow menu

### PR DESCRIPTION
**Pull request by jdandrade**
Overflow menu in all cards.

  Overflow button options:
1- delete post
2- report abuse
3- unfollow store/user

https://aptoide.atlassian.net/browse/AN-2016

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialMediaViewHolder.java [hunk 1]
```diff
@@ -309,4 +310,48 @@ private void showNumberOfLikes(long likesNumber) {
...
+  private void setupOverflowMenu(SocialMedia post, int position) {
+    overflowMenu.setOnClickListener(view -> {
+      PostPopupMenuBuilder postPopupMenuBuilder = new PostPopupMenuBuilder();
+      postPopupMenuBuilder.prepMenu(itemView.getContext(), overflowMenu)
+          .addReportAbuse(menuItem -> {
+            cardTouchEventPublishSubject.onNext(
+                new CardTouchEvent(post, position, CardTouchEvent.Type.REPORT_ABUSE));
+            return false;
+          });
+      accountManager.accountStatus()
```
Comment (marcelorbenites) [inline app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialMediaViewHolder.java]: AptoideAccountManager to to model layer. To make it simple you can query account state every time Posts are requested. Ideally the Posts that require account status would change every time the account changes.

File: app/src/main/java/cm/aptoide/pt/social/data/CardViewHolderFactory.java [hunk 2]
```diff
@@ -42,19 +44,24 @@
   private final MinimalCardViewFactory minimalCardViewFactory;
   private final String marketName;
   private final TimelineAdsRepository adsRepository;
+  private final AptoideAccountManager accountManager;
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/data/CardViewHolderFactory.java]: fixed, removed account manager dependencies from the viewholders.

File: app/src/main/java/cm/aptoide/pt/social/data/MinimalCardViewFactory.java [hunk 3]
```diff
@@ -309,6 +317,41 @@ private void showNumberOfLikes(long likesNumber, Context context) {
...
+      View subCardView) {
+    overflowMenu.setOnClickListener(view -> {
+      PostPopupMenuBuilder postPopupMenuBuilder =
+          new PostPopupMenuBuilder().prepMenu(subCardView.getContext(), overflowMenu)
+              .addReportAbuse(menuItem -> {
+                cardTouchEventPublishSubject.onNext(
+                    new CardTouchEvent(post, position, CardTouchEvent.Type.REPORT_ABUSE));
+                return false;
+              });
+      accountManager.accountStatus()
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/data/MinimalCardViewFactory.java]: fixed with the removal of the account manager dependency.

File: app/src/main/java/cm/aptoide/pt/social/data/publisher/Poster.java [hunk 4]
```diff
@@ -78,4 +78,11 @@ public String getSecondaryAvatar() {
   public String getSecondaryName() {
     return secondaryName;
   }
+
+  public boolean isMe(String username, String storeName) {
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/data/publisher/Poster.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialMediaViewHolder.java (hunk 1 above)
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialMediaViewHolder.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialPostMediaViewHolder.java [hunk 5]
```diff
@@ -322,4 +322,48 @@ private void showNumberOfLikes(long likesNumber) {
...
+                return false;
+              });
+            }
+            if (post.getPoster() != null) {
+              if (post.getPoster()
+                  .getUser() != null && !post.getPoster()
+                  .isMe(account.getNickname(), account.getStore()
+                      .getName())) {
+                postPopupMenuBuilder.addUnfollowUser(menuItem -> {
+                  cardTouchEventPublishSubject.onNext(new UserUnfollowCardTouchEvent(
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialPostMediaViewHolder.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialPostRecommendationViewHolder.java [hunk 6]
```diff
@@ -298,4 +297,48 @@ private void showNumberOfLikes(long likesNumber) {
...
+                return false;
+              });
+            }
+            if (post.getPoster() != null) {
+              if (post.getPoster()
+                  .getUser() != null && !post.getPoster()
+                  .isMe(account.getNickname(), account.getStore()
+                      .getName())) {
+                postPopupMenuBuilder.addUnfollowUser(menuItem -> {
+                  cardTouchEventPublishSubject.onNext(new UserUnfollowCardTouchEvent(
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialPostRecommendationViewHolder.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialRecommendationViewHolder.java [hunk 7]
```diff
@@ -91,93 +94,92 @@ public SocialRecommendationViewHolder(View view, int titleStringResourceId,
...
 
     this.getAppButton.setOnClickListener(click -> cardTouchEventPublishSubject.onNext(
-        new CardTouchEvent(card, position, CardTouchEvent.Type.BODY)));
+        new CardTouchEvent(post, position, CardTouchEvent.Type.BODY)));
     this.appIcon.setOnClickListener(click -> cardTouchEventPublishSubject.onNext(
-        new CardTouchEvent(card, position, CardTouchEvent.Type.BODY)));
+        new CardTouchEvent(post, position, CardTouchEvent.Type.BODY)));
     this.cardHeader.setOnClickListener(click -> cardTouchEventPublishSubject.onNext(
-        new SocialHeaderCardTouchEvent(card, card.getPoster()
+        new SocialHeaderCardTouchEvent(post, post.getPoster()
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialRecommendationViewHolder.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigation.java [hunk 8]
```diff
@@ -45,4 +45,6 @@
   void navigateToNotificationCenter();
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigation.java]: fixed

File: app/src/main/java/cm/aptoide/pt/social/data/UserUnfollowCardTouchEvent.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Created by jdandrade on 19/10/2017.
+ */
+
+public class UserUnfollowCardTouchEvent extends CardTouchEvent {
+  private final String name;
+  private final Long id;
+
+  public UserUnfollowCardTouchEvent(Long id, String name, int position, Post post) {
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/data/UserUnfollowCardTouchEvent.java]: To fix in the future 👍 
https://aptoide.atlassian.net/browse/AN-2077

File: app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigator.java [hunk 10]
```diff
@@ -140,4 +140,9 @@ public void navigateToAppView(long appId, String packageName, AppViewFragment.Op
   @Override public void navigateToCreatePost() {
     fragmentNavigator.navigateTo(PostFragment.newInstanceFromTimeline(), true);
   }
+
+  @Override public void navigateToFeedbackScreen(String path, String postId) {
```
[the remark this answers is not in the document]
Comment (jdandrade) [reply app/src/main/java/cm/aptoide/pt/social/presenter/TimelineNavigator.java]: fixed

**Pull request merged**